### ISSUE TITLE
Video player Issues

### ISSUE THREAD

**Issue by ghost** [Eclipse-Community/r3dfox-old#9]
![how-do-i-fix-this-i-am-using-betterfox-it-only-happens-on-a-v0-5q05g220wa2c1](https://github.com/Eclipse-Community/r3dfox/assets/155431995/c68b1c8c-614f-4668-a1e6-cbbe53b9dc47)
So for some reason everytime I tried to fastforward in aniwatch.to or 2movieorca or lookmovie2.to I get this visual glitch, note it doesn't happen with google chrome or any its forks and neither does it happen with pale moon, but it does happen with firefox or any of its forks, also some reels in instagram don't want to play, that issue only applies to windows 7, later windows don't suffer from that issue, strangely chrome does play them correctly, i would like to have your insight on that, if it is possible to help with that.

**Comment by the-r3dacted** [comment]
This would be an issue with either the Windows 7 media foundation or some issue with Firefox itself. I don't have the knowledge to tackle such a problem. I'll keep this open to hopefully get a pull request or something idk.

**Comment by jonm58** [comment]
> ![how-do-i-fix-this-i-am-using-betterfox-it-only-happens-on-a-v0-5q05g220wa2c1](https://private-user-images.githubusercontent.com/155431995/300907396-c68b1c8c-614f-4668-a1e6-cbbe53b9dc47.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2MzY4NDcsIm5iZiI6MTcwNjYzNjU0NywicGF0aCI6Ii8xNTU0MzE5OTUvMzAwOTA3Mzk2LWM2OGIxYzhjLTYxNGYtNDY2OC1hMWU2LWNiYmU1M2I5ZGM0Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEzMFQxNzQyMjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lOGI1ZmNkMGJhNDA5Nzc4MzJiOWFjYTlhODY0YjcxYzVmZGNiNjBjNmNjZjcxZjVmMTY0ZmU5ZjBiMjE5MWFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.7LoABh-72HKOUjXxG5jou-QwN3gTk4UHbau5_Hjqh10) So for some reason everytime I tried to fastforward in aniwatch.to or 2movieorca or lookmovie2.to I get this visual glitch, note it doesn't happen with google chrome or any its forks and neither does it happen with pale moon, but it does happen with firefox or any of its forks, also some reels in instagram don't want to play, that issue only applies to windows 7, later windows don't suffer from that issue, strangely chrome does play them correctly, i would like to have your insight on that, if it is possible to help with that.

Can I see your Decoder support information on the about:support page?

**Comment by ghost** [comment]
[Waterfox Log.txt](https://github.com/Eclipse-Community/r3dfox/files/14102434/Waterfox.Log.txt)
Quick note that the issue with video player does occur in Windows 10 at last on my laptop but the Instagram Reels shows videos in Windows 10, but only some on Windows 7. 
Also my specs are as follows: Windows 7 Ultimate 64-bit SP1  Intel Core2 Quad CPU Q8400 @ 2.66GHz, 2,0GB RAM,  AMD Radeon HD 6670.

**Comment by jonm58** [comment]
> [Waterfox Log.txt](https://github.com/Eclipse-Community/r3dfox/files/14102434/Waterfox.Log.txt) Quick note that the issue with video player does occur in Windows 10 at last on my laptop but the Instagram Reels shows videos in Windows 10, but only some on Windows 7. Also my specs are as follows: Windows 7 Ultimate 64-bit SP1 Intel Core2 Quad CPU Q8400 @ 2.66GHz, 2,0GB RAM, AMD Radeon HD 6670.

you use waterfox,please to waterfox github report:
https://github.com/WaterfoxCo/Waterfox

**Comment by ghost** [comment]
Yeah, but like I said even when I use R3dfox fork that issue still persist case in point
[download_01-30-20_22_59.webm](https://github.com/Eclipse-Community/r3dfox/assets/155431995/94f232ce-a50b-4de1-8a10-d645c2267c06)
 
[download_01-30-20_23_56.webm](https://github.com/Eclipse-Community/r3dfox/assets/155431995/6de084cc-1ac7-4af1-8548-00ced4d6552c)

**Comment by the-r3dacted** [comment]
You are more likely to get a solution reporting this on bugzilla or Waterfox GitHub.

**Comment by jonm58** [comment]
Well, "Codec Support Information:" in your log is not normal
You are more likely to get a solution reporting this on bugzilla or Waterfox GitHub.
![屏幕捕获_2024_01_31_02_33_05_878](https://github.com/Eclipse-Community/r3dfox/assets/73036321/38e005e4-8dc5-4d03-a1ed-70a5b1ed35e8)

**Comment by ghost** [comment]
Yeah, maybe but could you look at Nightly build log I've just download from this repository 
[Nightly Log.txt](https://github.com/Eclipse-Community/r3dfox/files/14102860/Nightly.Log.txt)
Also again just for quick reference same issue
 
[download_01-30-20_39_42.webm](https://github.com/Eclipse-Community/r3dfox/assets/155431995/858bb22a-fd81-45bf-b9bf-a22ca903ce94)

**Comment by jonm58** [comment]
@K4sum1 
I Love AMD(not)
![屏幕捕获_2024_01_31_02_53_49_653](https://github.com/Eclipse-Community/r3dfox/assets/73036321/69a4f078-aff8-427c-93d0-3a2a6691bde8)

**Comment by jonm58** [comment]
```
Description: AMD Radeon HD 6670
Vendor ID: 0x1002
Device ID: 0x6758
Driver Version: 8.982.0.0
Driver Date: 7-27-2012
Drivers: aticfx64 aticfx64 aticfx64 aticfx32 aticfx32 aticfx32 atiumd64 atidxx64 atidxx64 atiumdag atidxx32 atidxx32 atiumdva atiumd6a atitmm64
Subsys ID: 23091787
RAM: 1024
GPU #2
Active: No
RAM: 0
Diagnostics
AzureCanvasBackend: skia
AzureContentBackend: skia
AzureFallbackCanvasBackend: skia
CMSOutputProfile: 10204 bytes, too large
Display0: 1920x1080@60Hz scales:1.000000|1.000000
DisplayCount: 1
HardwareStretching: both=0 window-only=0 full-screen-only=0 none=0 error=1
ClearType Parameters: Gamma: 2.2 Pixel Structure: RGB ClearType Level: 100 Enhanced Contrast: 50
Decision Log
HW_COMPOSITING:
default: available,
D3D11_COMPOSITING:
default: available,
DIRECT2D:
default: available,
env: blocklisted, #BLOCKLIST_FEATURE_FAILURE_DL_BLOCKLIST_g144, Blocklisted; failure code FEATURE_FAILURE_DL_BLOCKLIST_g144
D3D11_HW_ANGLE:
default: available,
GPU_PROCESS:
default: available,
env: unavailable, GPU Process requires the Windows 7 Platform Update, Blocklisted; failure code FEATURE_FAILURE_PLATFORM_UPDATE
WEBRENDER:
default: available,
WEBRENDER_COMPOSITOR:
default: available,
runtime: unavailable, No DirectComposition usage, Blocklisted; failure code FEATURE_FAILURE_NO_GPU_PROCESS
WEBRENDER_PARTIAL:
default: available,
WEBRENDER_SHADER_CACHE:
default: available,
WEBRENDER_OPTIMIZED_SHADERS:
default: available,
WEBRENDER_ANGLE:
default: available,
WEBRENDER_DCOMP_PRESENT:
default: available,
env: unavailable, Requires GPU process, Blocklisted; failure code FEATURE_FAILURE_NO_GPU_PROCESS
WEBRENDER_SCISSORED_CACHE_CLEARS:
default: available,
WEBGPU:
default: available,
runtime: blocked, WebGPU cannot be enabled in release or beta, Blocklisted; failure code WEBGPU_DISABLE_RELEASE_OR_BETA
WINDOW_OCCLUSION:
default: available,
VP8_HW_DECODE:
default: available,
VP9_HW_DECODE:
default: available,
REUSE_DECODER_DEVICE:
default: available,
BACKDROP_FILTER:
default: available,
CANVAS_RENDERER_THREAD:
default: available,
ACCELERATED_CANVAS2D:
default: disabled, Disabled by default, Blocklisted; failure code FEATURE_FAILURE_DISABLED
runtime: failed, Disabled with non-zero canvas worker threads, Blocklisted; failure code FEATURE_FAILURE_DISABLE_BY_CANVAS_WORKER_THREADS
REMOTE_CANVAS:
default: available,


Failure Log
(#0) Error: shader-cache: Timed out before finishing loads

**Comment by ghost** [comment]
Sorry about I'm total amateur at such stuff,  didin't know I had to run media in order to show codec support anyway here are correct logs this time 
[Nightly Log.txt](https://github.com/Eclipse-Community/r3dfox/files/14103023/Nightly.Log.txt)
[WaterFox Log.txt](https://github.com/Eclipse-Community/r3dfox/files/14103025/WaterFox.Log.txt)

**Comment by jonm58** [comment]
> 抱歉，我对这些东西完全是业余的，不知道我必须运行媒体才能显示编解码器支持，无论如何，这次是正确的日志 [夜间日志.txt ](https://github.com/Eclipse-Community/r3dfox/files/14103023/Nightly.Log.txt) [水狐日志.txt ](https://github.com/Eclipse-Community/r3dfox/files/14103025/WaterFox.Log.txt)

CPU Video times
lmao your GPU don't working
You install Windows 7 Platform Update?

**Comment by jonm58** [comment]
https://wiki.mozilla.org/QA/Platform/Graphics/Features/GPU_Process
https://support.microsoft.com/en-us/topic/platform-update-for-windows-7-sp1-and-windows-server-2008-r2-sp1-d97da9ca-c15c-b21f-ebb0-838f7be8d9f6#:~:text=Property%20Value%20%20System%20%20%20%20Graphics,8.88%2B2342%2C%20A02%2C%20A03%20%28Released%203%2F2012%29%208.%20...%20

**Comment by jonm58** [comment]
@BurningDalylight https://www.microsoft.com/en-us/download/details.aspx?id=36805

**Comment by ghost** [comment]
So in nutshell I just don't have update support for Windows 7 huh, then a question why don't I have the issue with pale moon browser but I guess they use old engine Goanna, so probably that's the  reason but why don't have the same issues with chrome browser, but I guess by the end of today it just how it works.

**Comment by jonm58** [comment]
Pale Moon:No GPU Process
Chrome:have money or open source Codec lol

I want to be able to use ffmpeg, like in Firefox on linux

**Comment by garry-ut99** [comment]
I use Win7 SP1, despite I've switched today from FF 115 ESR to RF 124.0.2, I still have the same issue as described in : `https://github.com/uBlockOrigin/uBlock-issues/issues/3138#issuecomment-2016777561` - a video embeded in a github comment won't play.

Video broken in:
- FF 115 ESR + Win 7 SP1
- RF 124.0.2 + Win 7 SP1
- Opera 95 + Win 7 SP1
- Edge 109 + Win 7 SP1

Video plays fine on:
- FF 124 + Win 10 22H2
- RF 124 + Win 10 22H2
- Edge 123 + Win 10 22H2
- FF 124 + Ubuntu 23.10 
- FF 124 Mobile + Android 10
- Supremium 122 + Win 7 SP1

The bug affects all browsers, but in Win 7 SP1 only, which means the bug is specific for Win 7 SP1, but they fixed it in Supremium 122 for Win 7 SP1, so maybe possible to fix it in R3dfox for Win 7 SP1 somehow as well, maybe you can ask on Supremium repository how they fixed it or find which piece of code fixed it in Supremium, so you can implement the same fix into R3dfox.

Or maybe it's just some video codec which is neccesary to play the video, and which they embedded / implemented in Supremium, but is not embedded / implemented in Firefox, and Win 10 does have it, while Win 7 SP1 does not.

Or maybe my Win7 SP1 just lacks some codec / update.

**Comment by the-r3dacted** [comment]
> maybe you can ask on Supremium repository how they fixed it or find which piece of code fixed it in Supremium

As I am effectively banned (permanently muted) from the Supermium server, I doubt they would just give me that information.

If you can figure out which version fixed it, I could check the commits between them.

**Comment by garry-ut99** [comment]
Thx for instant reply and reopening the issue, I've checked old versions of Supremium beginning from the first Supremium version released: v115 and the bug was present in all v115 releases, then I've installed this v117 release: https://github.com/win32ss/supermium/releases/tag/v117 (using  "mini_installer.exe") and the video plays fine.

Looking at the change log of Supremium 117 version `117.0.5859.0`, and trying to guess - maybe it's this change which fixed the bug (but I'm only guessing):

> H.264 decoder (used for YouTube live streams, Twitch etc.)

**Comment by the-r3dacted** [comment]
So I tried looking here, and didn't see anything in the top 9 commits.

https://github.com/win32ss/supermium/commits/5c3b1753de3d9de87fa7673c2bc83171217e7675

I think ok, scrolling though before merge would be too much work, I can just search by user, and I try that.

https://github.com/win32ss/supermium/commits/5c3b1753de3d9de87fa7673c2bc83171217e7675?author=win32ss

No commits history.

I try the same on the main branch and still no commit history.

https://github.com/win32ss/supermium/commits?author=win32ss

I try private mode in case it's some weird block feature. Nothing there either.

I don't know where to look.

I assume this is from adding back XP/Vista support which has a limited or no media foundation. Not sure what to do here if anything since I don't really need a massive adding H264 decoder commit increasing code complexity even more to fix a one off problem.

**Comment by garry-ut99** [comment]
> I don't know where to look.

Supremium repo has almost 1,300,000 commits, github sometimes fails to `blame` or to return some results if a repository has too many commits, I've used a calendar on the right of github site to limit the timeframe and it has worked, so maybe you will find something interesting here. : https://github.com/win32ss/supermium/commits/v117?author=win32ss&since=2023-05-12&until=2024-03-12 

> I assume this is from adding back XP/Vista support which has a limited or no media foundation. Not sure what to do here if anything since I don't really need a massive adding H264 decoder commit increasing code complexity even more to fix a one off problem.

> to fix a one off problem.

Whether one or not, sometimes it's quality and not quantity that counts: a bug preventing playing some videos on very popular websites like GitHub is critical to be fixed in my opinion, it seems creators of Supremium were the same opinion, that's why they fixed it.

However, if you are gonna choose to not fix it, then could you at least provide a guide how to fix it on user side, so that I can fix it on my side, maybe it's sufficient to just put some codec file in the r3dfox folder or install codec in the system, or maybe you can even create some separate patch for users who want to install it and fix the bug on their side.

**Comment by jonm58** [comment]
> So I tried looking here, and didn't see anything in the top 9 commits.
> 
> https://github.com/win32ss/supermium/commits/5c3b1753de3d9de87fa7673c2bc83171217e7675
> 
> I think ok, scrolling though before merge would be too much work, I can just search by user, and I try that.
> 
> https://github.com/win32ss/supermium/commits/5c3b1753de3d9de87fa7673c2bc83171217e7675?author=win32ss
> 
> No commits history.
> 
> I try the same on the main branch and still no commit history.
> 
> https://github.com/win32ss/supermium/commits?author=win32ss
> 
> I try private mode in case it's some weird block feature. Nothing there either.
> 
> I don't know where to look.
> 
> I assume this is from adding back XP/Vista support which has a limited or no media foundation. Not sure what to do here if anything since I don't really need a massive adding H264 decoder commit increasing code complexity even more to fix a one off problem.

add ffmpeg dll support?

**Comment by the-r3dacted** [comment]
Can you check if Mypal68 has the same issue? I assume it has some media foundation workaround for XP, and maybe that plays the video.

https://codeberg.org/Theodor2/Mypal68/releases

**Comment by garry-ut99** [comment]
I've checked and the video doesn't play in Mypal68 (`68.13.9b`) in WIN 7 SP1 (but it does play in Win 10 22H2).

**Comment by the-r3dacted** [comment]
Hmm, can you try in a XP VM? Or just send the vid so I can check?

**Comment by garry-ut99** [comment]
In my first comment https://github.com/Eclipse-Community/r3dfox/issues/9#issuecomment-2028162510 I already provided a link :

> garry-ut99 :  I use Win7 SP1, despite I've switched today from FF 115 ESR to RF 124.0.2, I still have the same issue as described in : `https://github.com/uBlockOrigin/uBlock-issues/issues/3138#issuecomment-2016777561` - a video embeded in a github comment won't play.

to a comment which contains a link to another comment which contains the video (so you can test the video yourself):  `https://github.com/uBlockOrigin/uBlock-issues/issues/3138#issuecomment-1962793859`

Anyway I've checked myself on Win XP SP3 in Mypal 68 and the video plays fine.

**Comment by the-r3dacted** [comment]
Ah you put the link as a code. I was scanning the page for blue links and therefore didn't see that. Since it plays fine in XP, I think it might be easier to figure out the video playback code for Mypal68 and try porting it to this browser.

**Comment by garry-ut99** [comment]
> Ah you put (...)

Yes, I did that to avoid unnecessary cross-referencing to unrelated issue in unrelated repository.

**Comment by Stepman123** [comment]
![play-mp3](https://github.com/Eclipse-Community/r3dfox/assets/8846337/9941814e-87de-43b1-9bf2-0af51e465f17)

I have a similar problem. If I open an mp3 file from a disk, I see a player that does not start playing.
If I open an mp4, I see the error "...file is corrupt" like in the screenshots above.
YouTube also cannot start playing.
However, the problem only exists on Server 2003 with OCA 32-bit. On Windows 7 32-bit, on the same PC there are no problems. Video and sound work.

In browsers based on chromium ~(109-123) with OCA, the video plays, but without sound. However, this problem is easily resolved by adding key --enable-exclusive-audio

I thought that there should be a similar key or option for Firefox-based browsers.
But it looks like there is no tricky way?

**Comment by jonm58** [comment]
> ![play-mp3](https://private-user-images.githubusercontent.com/8846337/324202272-9941814e-87de-43b1-9bf2-0af51e465f17.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM2NDQwOTUsIm5iZiI6MTcxMzY0Mzc5NSwicGF0aCI6Ii84ODQ2MzM3LzMyNDIwMjI3Mi05OTQxODE0ZS04N2RlLTQzYjEtOWJmMi0wYWY1MWU0NjVmMTcuUE5HP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjBUMjAwOTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2E1NjBhMjYzNmI0MmUwMzIxMzU0NDZlNjlmMDQ2NjJmY2RlODk5NmE1MjI5YmYzMjE5YjFmNzYzMGFkYTYxNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.yfyKjX2xhXfTMBcp-JsFx4kBtichIxFS9FkDgw-raBY)
> 
> I have a similar problem. If I open an mp3 file from a disk, I see a player that does not start playing. If I open an mp4, I see the error "...file is corrupt" like in the screenshots above. YouTube also cannot start playing. However, the problem only exists on Server 2003 with OCA 32-bit. On Windows 7, on the same PC there are no problems. Video and sound work.
> 
> In browsers based on chromium ~(109-123) with OCA, the video plays, but without sound. However, this problem is easily resolved by adding key --enable-exclusive-audio
> 
> I thought that there should be a similar key or option for Firefox-based browsers. But it looks like there is no tricky way?

https://github.com/Feodor2/Mypal68/issues/423
You can test vp8 and vp9?
maybe only vp8 and vp9 working
~~edit:OK,in XP can't watch any video~~
edit:OneCoreAPI lost Environment variables.

**Comment by Stepman123** [comment]
> maybe only vp8 and vp9 working

No, these files don't work either. I tried from here https://test-videos.co.uk/jellyfish/webm-vp8
These files don't contain audio, but they still don't work.

**Comment by the-r3dacted** [comment]
I know Firefox uses the media foundation which XP doesn't have. I wonder how video in Chrome works on OCAPI. I'm also pretty sure VP8/VP9 should be internal codec and not media foundation but idk. Try AV1 maybe?

**Comment by jonm58** [comment]
> Try AV1 maybe?

Firefox H.264\H.265\AV1 use the media foundation,OCAPI Firefox 115ESR VP8/9 maybe work
edit:you can in about:support see your Codec Support Information
![image](https://github.com/Eclipse-Community/r3dfox/assets/73036321/e9a16d15-34c5-475e-aa2e-e8c575db6e3c)

**Comment by garry-ut99** [comment]
Is hardware VP9 / AV1 decoding support going to be implemented in R3dfox as per https://github.com/Eclipse-Community/r3dfox/issues/34#issuecomment-2068150765

**Comment by the-r3dacted** [comment]
Not sure. Not something I have a way of testing either unless VMware Workstation 16 can decode them.

**Comment by garry-ut99** [comment]
I've tested on VMware 15 + Win 10 22H2 + R3dfox 125 and it detects hardware decoding properly:

![codecs](https://github.com/Eclipse-Community/r3dfox/assets/72945564/08171163-c316-4d16-a7cd-0b8ec79d5233)

Althought when trying to play VP9 video on YouTube, hardware decoding doesn't work and both CPU & GPU are much occupied.

But isn't sufficient for you to just have hardware decoding flags marked as green in about:support in your Win 7 + VM, then I can do real test part on my side as I have real Win 7 SP1 as soon as you make these flags to appear as green in your VM.

**Comment by Stepman123** [comment]
> Firefox H.264\H.265\AV1 use the media foundation,OCAPI Firefox 115ESR VP8/9 maybe work
> edit:you can in about:support see your Codec Support Information

What system was the screenshot taken on?

**Comment by jonm58** [comment]
> > Firefox H.264\H.265\AV1 use the media foundation,OCAPI Firefox 115ESR VP8/9 maybe work
> > edit:you can in about:support see your Codec Support Information
> 
> What system was the screenshot taken on?

Win11

**Comment by Stepman123** [comment]
In Windows 7 it looks like this:
![win7Codec](https://github.com/Eclipse-Community/r3dfox/assets/8846337/b49c4f19-8b84-4c8b-9cc1-655c5db00b65)

2003 OCA:
![2003Codec](https://github.com/Eclipse-Community/r3dfox/assets/8846337/89eb7bad-cc58-4fbb-aefe-3817a6db6353)

**Comment by Stepman123** [comment]
On Windows 7 it detects incorrectly. In fact, software decoding is supported for all variants.

**Comment by garry-ut99** [comment]
Yeah, for me in Win 7 + R3dfox 125.0.2 same, most of Software is "Unsupported"
but then I modified some flags in `about:config` and almost all  Software turned to "Supported":

Fresh browser profile:

![1](https://github.com/Eclipse-Community/r3dfox/assets/72945564/8312f5e3-32ac-41c1-84c3-a0947adf783d)

After modification:

![2](https://github.com/Eclipse-Community/r3dfox/assets/72945564/6676ae85-a223-491f-b99e-beff4ad357d3)

**Comment by the-r3dacted** [comment]
What flags? Also what does 115 ESR look like?

**Comment by garry-ut99** [comment]
I undo my statement, it was not changed flags what fixed it for me, I got wrongly misleaded and fooled by noob Stepman123's suggestion :

> Stepman123 : ( https://github.com/Eclipse-Community/r3dfox/issues/9#issuecomment-2073362923 ) :  In Windows 7 it looks like this:
> Stepman123 ( https://github.com/Eclipse-Community/r3dfox/issues/9#issuecomment-2073384672 ): On Windows 7 it detects incorrectly. In fact, software decoding is supported for all variants.

Not incorrectly, it's by design, on your second screenshoot it says: 

> Codec support information unavailable. Try again after playing back a media file.

> https://askubuntu.com/questions/1480436/hardware-acceleration-in-firefox-not-working-after-recent-update
> 
> PS: It looks like this section is not always present.
> It says: Codec support information unavailable. Try again after playing back a media file.
> So, do what it says, play a media file, reload the about:support page, and voila.

The same applies for your first screenshoot as well:

> https://www.reddit.com/r/firefox/comments/18nh89l/is_it_normal_on_windows_10_that_so_many_codecs/ :
>
> They're not unsupported, you need to play video or audio first and then those will appear as supported after you refresh the about:support page. I'm guessing Firefox doesn't check what codecs are supported until after you've used media playback of some kind. Just opening a YouTube video should work. 
>
> SOLVED: The software decoders don't get detected as soon as you open Firefox. You need to visit a Web site first. Not all Web sites trigger the change, but YouTube and Twitch do. Visit one of those Web sites (no need to view a video) and refresh the about:support page and now all the software decoders will change to [supported](https://i.imgur.com/PTzl405.jpg).

**Comment by Stepman123** [comment]
> Codec support information unavailable. Try again after playing back a media file.

It doesn't really work as described on the internet. On Windows 7 yes, just play a media file and almost everyone turns green.
On 2003 it doesn't work. Still "Codec support information unavailable. Try again after playing back a media file."
I knew about this trick before I wrote the post, but I haven't tried it on Windows 7.

**Comment by jonm58** [comment]
![image](https://github.com/Eclipse-Community/r3dfox/assets/73036321/86b82b58-a456-4154-8b28-a8776d06289c)
@Stepman123 
I testing your vp9 video,i'm use **Environment variables**:
MOZ_DISABLE_CONTENT_SANDBOX=1
MOZ_DISABLE_GMP_SANDBOX=1
**MOZ_DISABLE_RDD_SANDBOX=1**
MOZ_DISABLE_SOCKET_PROCESS_SANDBOX=1
edit:AV1 and vp8/9 use Data Decoder process?

**Comment by jonm58** [comment]
> ![image](https://private-user-images.githubusercontent.com/73036321/326461797-86b82b58-a456-4154-8b28-a8776d06289c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ0MDE2MDIsIm5iZiI6MTcxNDQwMTMwMiwicGF0aCI6Ii83MzAzNjMyMS8zMjY0NjE3OTctODZiODJiNTgtYTQ1Ni00MTU0LThiMjgtYTg3NzZkMDYyODljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDE0MzUwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlYmYyNDllZDk2NTlhODFhYjkwMDg4M2Y0MTYzMmIwMDE0NDk2NTlmZTE5N2E0ZmM5MGUyMTJhODE4MTRiNTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.nE4A_NHs753x96imBjKUirjjqXqRGYXD4gTk1X-uDh0) @Stepman123 I testing your vp9 video,i'm use **Environment variables**: MOZ_DISABLE_CONTENT_SANDBOX=1 MOZ_DISABLE_GMP_SANDBOX=1 **MOZ_DISABLE_RDD_SANDBOX=1** MOZ_DISABLE_SOCKET_PROCESS_SANDBOX=1 edit:vp8/9 use Data Decoder process?

AV1(webm and mp4) and VP8/9 working
H.264 and H.265(HEVC) don't work

**Comment by jonm58** [comment]
Mozilla FireFox ESR 115.5.0 on Windows2003 One Core API youtube video working
![1.png](https://cdn.discordapp.com/attachments/1030000423282675714/1234543827675643995/image.png?ex=66311e09&is=662fcc89&hm=aa5ee293de7af5b04e2b83ebd27271ecc2071f3a372be8e0e401cc6701a3e2dc&)

but Eclipse r3dfox 125.0.3 on Windows2003 One Core API can't
(All test on VMware)
[test video link](https://www.youtube.com/watch?v=HR4REfXv0j8)
[r3dfox 125.0.3 youtube profiler](https://share.firefox.dev/4aUakC5)

**Comment by jonm58** [comment]
Disable e10s,break video player.
![屏幕截图 2024-06-16 050659](https://github.com/Eclipse-Community/r3dfox/assets/73036321/427b0e0d-38e8-4cf6-a686-c706efb3a1c7)

**Comment by garry-ut99** [comment]
### 1

> https://github.com/Eclipse-Community/r3dfox/issues/9#issuecomment-2028162510: garry-ut99  : a video embeded in a github comment won't play.
>
> a bug preventing playing some videos on very popular websites like GitHub is critical to be fixed in my opinion, it seems creators of Supremium were the same opinion, that's why they fixed it.

> https://github.com/Eclipse-Community/r3dfox/issues/9#issuecomment-2028243317 : K4sum1: I don't really need a massive adding H264 decoder commit increasing code complexity even more to fix a one off problem.

It's a terrible experience to not be able to watch some videos because you don't like to embed the required video coded.

<hr>

### 2

Yet another terrible experience was, when you cut-off Firefox Profiler, with a strange reason:

> https://github.com/Eclipse-Community/r3dfox/issues/86 : https://github.com/Eclipse-Community/r3dfox/commit/83b286649bd3e984857ffdf73fba1d2a8a59ad00 : K4sum1 : Profiler isn't very likely to be used

Fortunately you bringed back the Firefox Profiler after I complained.

<hr>

### 3

Now, another terrible experience is, when I've been trying to play videos on big and popular polish website `player.pl` : https://player.pl/programy-online/fakty-odcinki,516/odcinek-9686,S45E9686,274009, but it fails because R3dfox doesn't support DRM. Then I've found similiar issues, where people are complaing they can't play videos on many popular websites like Spotify, Netflix, related DRM issues:
- https://github.com/Eclipse-Community/r3dfox/issues/89
- https://github.com/Eclipse-Community/r3dfox/issues/72
- https://github.com/Eclipse-Community/r3dfox/issues/71
- https://github.com/Eclipse-Community/r3dfox/issues/58
- https://github.com/Eclipse-Community/r3dfox/issues/47
- https://github.com/Eclipse-Community/r3dfox/issues/50

> https://github.com/Eclipse-Community/r3dfox/issues/47#issuecomment-2059730690 : K4sum1 : I want to just fully disable it as I have no way to test, nor do I want closed source Google code running in my browser.

> https://github.com/Eclipse-Community/r3dfox/issues/72#issuecomment-2130212632 : K4sum1 : I don't want DRM running in my browser, nor do I have a way to test if it's broken

That is a terrible experience, to be unable to play videos from many popular websites, just because you don't like DRM.

<hr>

The more time goes, the more R3dfox becomes your private fork which suits your needs, where you skip important features which you don't like, but which can be still important for community, the two remaining video related issues: "lack of  video codec" + "lack of video DRM support" make R3dfox a broken browser for watching videos, because too many videos are broken, currently I need to use 2 browsers simultaneously: R3dfox, and additionally Supremium just to watch videos, it's getting uncomfortable. I wait for a day when Youtube videos stop loading in R3dfox...

I still do understand your arguments, regarding why you don't want to embed big codecs or to support DRM owned by Google, and that you might not have enough time or knowledge, I just wanted to share some feedback, the final decision still belongs to you, as you are the main maintainer.

**Comment by the-r3dacted** [comment]
> It's a terrible experience to not be able to watch some videos because you don't like to embed the required video coded.

I've changed my opinion. I would add it, but I don't know how, and I don't have the time or skill to do it myself. If someone were to make a pr for it, and it was stable and worked well, I'd merge it.

> That is a terrible experience, to be unable to play videos from many popular websites, just because you don't like DRM.

I should see if I can get @Alex313031 to base Mercury-Win7 off of r3dfox again. He does builds with DRM, and I could just point to that instead of adding DRM here.

**Comment by the-r3dacted** [comment]
@garry-ut99 Try this, it should be compiled with DRM

https://github.com/lunanakaya/luminafox/releases/tag/128.0.2-rc1

**Comment by the-r3dacted** [comment]
I'm not going to transfer this issue to the new repo as it seems likely to also be an issue in [Firefox-for-windows-7](https://github.com/e3kskoy7wqk/Firefox-for-windows-7/) and should be reported there instead.

If this happens to be a r3dfox only bug, make a [new issue in the new repo](https://github.com/Eclipse-Community/r3dfox/issues).
